org.eclipse.scanning.test/src/org/eclipse/scanning/test/command/ScanRequestCreationTest.java

```diff
@@ -143,6 +144,98 @@ public void testGridCommandWithROI() throws Exception {
+			ScanRequest<IROI> request = pi.get("sr", ScanRequest.class);
+
+			Collection<Object> models = request.getCompoundModel().getModels();
+			assertEquals(1, models.size());  // I.e. this is not a compound scan.
+
+			Object model = models.iterator().next();
+			assertEquals(RandomOffsetGridModel.class, model.getClass());
+
+			RandomOffsetGridModel rogmodel = (RandomOffsetGridModel) model;
+			assertEquals("fred", rogmodel.getFastAxisName());
+			assertEquals("y", rogmodel.getSlowAxisName());
+			assertEquals(5, rogmodel.getFastAxisPoints());
+			assertEquals(6, rogmodel.getSlowAxisPoints());
+			assertEquals(true, rogmodel.isSnake());
+			assertEquals(5, rogmodel.getOffset(), 1e-8);
+			assertEquals(0, rogmodel.getSeed());
+
+	}
+
+	@Test
+	public void testSpiralCommand() {
+		pi.exec("sr =                             	  "
+				+	"scan_request(                    "
+				+	"    spiral(			          "
+				+	"        axes=(my_scannable, 'y'),"
+				+	"        start=(0, 2),            "
+				+	"        stop=(10, 11),           "
+				+   "		 scale=(0.5),			  "
+				+	"        roi=circ((4, 6), 5)      "
+				+	"    )                            "
+				+	")                                ");
+			@SuppressWarnings("unchecked")
+			ScanRequest<IROI> request = pi.get("sr", ScanRequest.class);
+
+			Collection<Object> models = request.getCompoundModel().getModels();
+			assertEquals(1, models.size());  // I.e. this is not a compound scan.
+
+			Object model = models.iterator().next();
+			assertEquals(SpiralModel.class, model.getClass());
+
+			SpiralModel spiral = (SpiralModel) model;
+			assertEquals(0.5, spiral.getScale(), 1e-8);
+	}
+
+	@Test
+	public void testLissajousCommand() {
+		pi.exec("sr =                             	  "
+				+	"scan_request(                    "
+				+	"    lissajous(			          "
+				+	"        axes=(my_scannable, 'y'),"
+				+	"        start=(0, 2),            "
+				+	"        stop=(10, 11),           "
+				+   "		 a=(0.5),			 	  "
+				+   "		 b=(0.1),			 	  "
+				+   "		 delta=(5),			 	  "
+				+   "		 theta=(3),			 	  "
+				+   "		 points=(40),		 	  "
+				+	"        roi=circ((4, 6), 5)      "
+				+	"    )                            "
+				+	")                                ");
+			@SuppressWarnings("unchecked")
+			ScanRequest<IROI> request = pi.get("sr", ScanRequest.class);
+
+			Collection<Object> models = request.getCompoundModel().getModels();
+			assertEquals(1, models.size());  // I.e. this is not a compound scan.
+
+			Object model = models.iterator().next();
+			assertEquals(LissajousModel.class, model.getClass());
+
+			LissajousModel lissajous = (LissajousModel) model;
+			assertEquals(0.5, lissajous.getA(), 1e-8);
+			assertEquals(0.1, lissajous.getB(), 1e-8);
+			assertEquals(5, lissajous.getDelta(), 1e-8);
+			assertEquals(3, lissajous.getThetaStep(), 1e-8);
+			assertEquals(40, lissajous.getPoints());
+	}
+
+	@Test
 	public void testGridCommandWithROINoDetector() throws Exception {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)